src/extractKey.ts

```diff
@@ -0,0 +1,83 @@
+import { createOperation, Operation } from 'apollo-link';
+import { DirectiveNode, ArgumentNode } from 'graphql';
+
+import deepUpdate = require('deep-update');
```

Comment:
I don't really like adding new dependencies, especially if they have transitive dependencies (of which lodash is one in this case). Stripping a directive from a query is a very common use case for a link. Let's put that in apollo-link, or something like apollo-link-utils, if it doesn't already exist? cc @evans @jbaxleyiii